build.rs

```diff
@@ -0,0 +1,7 @@
+use rustc_version::{version_meta, Channel};
+
+fn main() {
+    if version_meta().unwrap().channel == Channel::Nightly {
+        println!("cargo:rustc-cfg=nightly_build");
```

Comment:
I prefer to introduce a new feature called `nightly` instead of adding a new dep in `build-dependencies`.

And I think the current implementation is wrong.

```diff
- #![cfg_attr(feature = "nightly_build", feature(build_hasher_simple_hash_one))]
+ #![cfg_attr(nightly_build, feature(build_hasher_simple_hash_one))]
```

Reply:
We can take `hashbrown`'s implementation for a look: https://github.com/rust-lang/hashbrown/pull/292

Reply:
> I prefer to introduce a new feature called nightly instead of adding a new dep in build-dependencies.

But users need to add this to the features list. `build.rs` can help to select automatically.

Reply:
> But users need to add this to the features list. `build.rs` can help to select automatically.

I prefer adding a new feature instead. Leave this for @jorgecarleitao to decide.

Reply:
I would also prefer a specific feature, but both are mergable :) - we use this for `simd` also where users activate simd if they are in nightly. Maybe the nightly feature can activate both simd and this path? I can work out the details once we merge this :)